File: app/src/main/java/com/noobfitness/noobfitness/MainActivity.java
package com.noobfitness.noobfitness;

import android.content.Intent;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        LinearLayout mainLinearLayout = (LinearLayout) findViewById(R.id.activity_main);

        Button fourDayDefault = new Button(this);
        fourDayDefault.setText("4-day default routine");
        fourDayDefault.setGravity(Gravity.LEFT);
        fourDayDefault.setPadding(96, 96, 96, 96);
        fourDayDefault.setTypeface(Typeface.create("sans-serif-light", Typeface.NORMAL));
        fourDayDefault.setTextSize(24);
        fourDayDefault.setAllCaps(false);
        fourDayDefault.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(v.getContext(), RoutineActivity.class);
                startActivity(i);
            }
        });
        mainLinearLayout.addView(fourDayDefault);

        TextView imageCredit = new TextView(this);
        imageCredit.setText("Image source: Everkinetic.com (CC BY-SA 3.0)");
        mainLinearLayout.addView(imageCredit);

    }

}

File: app/src/main/java/com/noobfitness/noobfitness/main/MainActivity.kt
package com.noobfitness.noobfitness.main

import android.app.Activity
import android.os.AsyncTask
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.noobfitness.noobfitness.R
import com.noobfitness.noobfitness.auth.AuthService
import com.noobfitness.noobfitness.auth.UserManager
import com.noobfitness.noobfitness.dagger.InjectedApplication
import com.squareup.picasso.Picasso
import okhttp3.OkHttpClient
import okhttp3.Request
import javax.inject.Inject

class MainActivity : Activity() {

    @Inject
    lateinit var authService: AuthService
    @Inject
    lateinit var userManager: UserManager

    private lateinit var makeApiCallButton: Button
    private lateinit var signOutButton: Button
    private lateinit var usernameTextView: TextView
    private lateinit var avatarImageView: ImageView

    private lateinit var userRoutineList: RecyclerView
    private lateinit var userRoutineAdapter: UserRoutineAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        (application as InjectedApplication).appComponent.inject(this)

        this.makeApiCallButton = findViewById(R.id.makeApiCall)
        this.signOutButton = findViewById(R.id.signOut)
        this.usernameTextView = findViewById(R.id.fullName)
        this.avatarImageView = findViewById(R.id.profileImage)
        this.userRoutineList = findViewById(R.id.userRoutinesList)

        makeApiCallButton.setOnClickListener { onMakeApiCallClicked() }
        signOutButton.setOnClickListener { onSignOutClicked() }
        usernameTextView.text = userManager.get()?.username
        Picasso.with(this)
                .load(userManager.get()?.avatar)
                .placeholder(R.drawable.ring_accent)
                .into(avatarImageView)
        this.userRoutineAdapter = UserRoutineAdapter()
        userRoutineList.adapter = userRoutineAdapter
        userRoutineList.layoutManager = LinearLayoutManager(this)

        userRoutineAdapter.userRoutines = userManager.get()?.routines!!
    }

    private fun onSignOutClicked() {
        userManager.set(null)
        authService.logout()
    }

    private fun onMakeApiCallClicked() {
        object : AsyncTask<String, Void, String>() {
            override fun doInBackground(vararg tokens: String): String? {
                val client = OkHttpClient()
                val request = Request.Builder()
                        .get()
                        .header("x-auth-token", tokens[0])
                        .url("http://10.0.2.2:5000/api/workouts")
                        .build()
                try {
                    val response = client.newCall(request).execute()
                    val jsonBody = response.body()!!.string()
                    return jsonBody
                } catch (exception: Exception) {
                    Log.w(LOG_TAG, exception)
                    return null
                }
            }

            override fun onPostExecute(jsonBody: String?) {
                Toast.makeText(this@MainActivity, jsonBody, Toast.LENGTH_LONG).show();
            }
        }.execute(userManager.get()?.authToken)
    }

    companion object {
        private val LOG_TAG = "MainActivity"
    }
}
